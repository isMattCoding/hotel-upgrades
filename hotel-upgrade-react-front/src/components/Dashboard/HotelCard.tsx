import HotelUpgrades from "./HotelUpgrades"
import { type HotelType } from "./HotelsList"

export default function HotelCard({hotel}: {hotel:HotelType}) {

  return <div className="w-full max-w-md p-4 bg-white border border-gray-200 rounded-lg shadow sm:p-8 dark:bg-gray-800 dark:border-gray-700">
    <div className="flex items-center justify-between mb-4">
      <h5 className="text-xl font-bold leading-none text-gray-900 dark:text-white">{hotel.name}, {hotel.city}</h5>
      <a href="/dashboard" className="text-sm font-medium text-blue-600 hover:underline dark:text-blue-500">
        View all
      </a>
    </div>
    <HotelUpgrades hotel_id={hotel.id} />
  </div>

}
