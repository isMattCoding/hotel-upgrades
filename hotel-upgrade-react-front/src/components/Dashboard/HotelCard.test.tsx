import React from 'react';
import { render, screen } from '@testing-library/react';
import HotelCard from './HotelCard';

const hotel = {
  id: 42,
  name: "Crown",
  city: "Melbourne"
}

test('renders h1 element', () => {
  const { container } = render(<HotelCard hotel={hotel}/>);
  expect(container).toMatchSnapshot()
  const h5Element = screen.getByText(`${hotel.name}, ${hotel.city}`)
  expect(h5Element).toBeInTheDocument();
});
