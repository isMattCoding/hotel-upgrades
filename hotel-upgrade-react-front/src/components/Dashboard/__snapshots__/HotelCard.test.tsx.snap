// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders h1 element 1`] = `
<div>
  <div
    class="w-full max-w-md p-4 bg-white border border-gray-200 rounded-lg shadow sm:p-8 dark:bg-gray-800 dark:border-gray-700"
  >
    <div
      class="flex items-center justify-between mb-4"
    >
      <h5
        class="text-xl font-bold leading-none text-gray-900 dark:text-white"
      >
        Crown
        , 
        Melbourne
      </h5>
      <a
        class="text-sm font-medium text-blue-600 hover:underline dark:text-blue-500"
        href="#"
      >
        View all
      </a>
    </div>
    Loading...
  </div>
</div>
`;
